bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -141,6 +141,11 @@ protected String preprocessSnippet(String originalSnippet, Widget w, boolean ign
         String state = getState(w);
         snippet = snippet.replace("%state%", escapeHtml(state));
         snippet = snippet.replace("%state_in_url%", escapeURL(state));
+        if ("UNDEF".equals(state) || "NULL".equals(state)) {
```

Comment:
If we want to be rigorous, it would be better to check itemUIRegistry.getState(w) and not the resulting string as you could have a valid item state for a string item with "NULL" state but which a NULL item state. 
I hope I am enough clear !

Reply:
Yes, good point. I will change it.